rocketpy/sensors/sensors.py

```diff
@@ -0,0 +1,373 @@
+        else:
+            raise ValueError("Invalid measurement range format")
+
+        # rotation matrix and normal vector
+        if any(isinstance(row, (tuple, list)) for row in orientation):  # matrix
+            self.rotation_matrix = Matrix(orientation)
+        elif len(orientation) == 3:  # euler angles
+            self.rotation_matrix = Matrix.transformation_euler_angles(
+                *orientation
+            ).round(12)
+        else:
+            raise ValueError("Invalid orientation format")
+        self.normal_vector = Vector(
+            [
+                self.rotation_matrix[0][2],
+                self.rotation_matrix[1][2],
+                self.rotation_matrix[2][2],
+            ]
+        ).unit_vector
+
+        # cross axis sensitivity matrix
+        _cross_axis_matrix = 0.01 * Matrix(
+            [
+                [100, self.cross_axis_sensitivity, self.cross_axis_sensitivity],
+                [self.cross_axis_sensitivity, 100, self.cross_axis_sensitivity],
+                [self.cross_axis_sensitivity, self.cross_axis_sensitivity, 100],
+            ]
+        )
+
+        # compute total rotation matrix given cross axis sensitivity
+        self._total_rotation_matrix = self.rotation_matrix @ _cross_axis_matrix
+
+        # map which rocket(s) the sensor is attached to and how many times
+        self._attached_rockets = {}
+
+    def __repr__(self):
+        return f"{self.name}"
+
+    def __call__(self, *args, **kwargs):
+        return self.measure(*args, **kwargs)
+
+    def _vectorize_input(self, value, name):
+        if isinstance(value, (int, float)):
+            return Vector([value, value, value])
+        elif isinstance(value, (tuple, list)):
+            return Vector(value)
+        else:
+            raise ValueError(f"Invalid {name} format")
+
+    def _reset(self, simulated_rocket):
+        """Reset the sensor data for a new simulation."""
+        self._random_walk_drift = Vector([0, 0, 0])
+        self.measured_data = []
+        if self._attached_rockets[simulated_rocket] > 1:
+            self.measured_data = [
+                [] for _ in range(self._attached_rockets[simulated_rocket])
+            ]
+            self._save_data = self._save_data_multiple
+        else:
+            self._save_data = self._save_data_single
+
+    def _save_data_single(self, data):
+        """Save the measured data to the sensor data list for a sensor that is
+        added only once to the simulated rocket."""
+        self.measured_data.append(data)
+
+    def _save_data_multiple(self, data):
+        """Save the measured data to the sensor data list for a sensor that is
+        added multiple times to the simulated rocket."""
+        self.measured_data[self._counter].append(data)
+        # counter for cases where the sensor is added multiple times in a rocket
+        self._counter += 1
+        if self._counter == len(self.measured_data):
+            self._counter = 0
```

Comment:
Sorry for bringing this up again, but the more I think about this, the more this structure worries me on maintainability in the future for the following reasons:

- Having two possible data types (list or list of lists) on the sensor measurement dict makes it rather tiring writing code for retrieving and analyzing these data (e.g. loads of boilerplate to check the list structure etc);
- This brings more code couplings, dependencies etc.

Was this an issue when developing the `Controllers`? Have we tried adding the same controller twice? IIRC, we did not consider that and would require essentially the same solution as here.

To be completely honest, right now my personal opinion is that we should not allow the same object to be added twice (the user could simply copy and paste the lines of code and create a new instance of the controller). However, we have already gone a lot into this problem and even implemented a working solution, so I would like to hear your opinion on the topics:

1. Do you foresee problems in future code dealing with these different data structures?
2. Does the `Controller` class (or any other future state saving class) suffers from the same issue (being added twice)?
    - If the answer is yes, should we implement this solution in all cases or disallow this operation?

Reply:
> 1. Do you foresee problems in future code dealing with these different data structures?

I don't see problems. Just some small if else statements

> Does the `Controller` class (or any other future state saving class) suffers from the same issue (being added twice)?
> * If the answer is yes, should we implement this solution in all cases or disallow this operation?

Controllers don't save data on themselves, so it does not encounter those problems

Reply:
Alright, thanks for your reply.

Do you consider the `observed_variables` as data saved on controllers?